src/main/java/io/codelex/java/advanced/test/Exercise2/Basket.java

```diff
@@ -0,0 +1,62 @@
+package io.codelex.java.advanced.test.Exercise2;
+
+import java.util.ArrayList;
+import java.util.List;
+
+public class Basket <T> {
+    private final int MAX_AMOUNT = 10;
+    private int amount;
+    private T item;
```

Comment:
Kāpēc grozs glabā tikai vienu lietu? Viņā vajadzētu varēt glabāt līdz pat 10 lietām, nevis tikai vienu.

Reply:
Šis uzdevums man likās visgrūtakais, jo es līdz galam tā arī nesapratu, kāda visam šim ir galvenā ideja.

#### NB! Basket can store only one kind of items - only apples, only mushrooms etc, it cannot store both apples and mushrooms at the same time. (not to cause confusion - it means that class Object cannot be used and must use generics)

Mana ideja bija tāda, ka tā kā grozs var glabāt tikai viena tipa <T> objektus, pietiek ar to, lai grozā glabājas tikai viens objekts un mēs vienkārši palielinam tā "skaitu". Jo abas Item klases ir tukšas un tām nav kādu specifisku īpašību pēc kurām tās var atšķirties (piemēram, pēc krāsas vai nosaukuma).